cps/tasks/download.py

```diff
@@ -111,6 +118,7 @@ def run(self, worker_thread):
                         # 2024-02-17: Dedup Design Evolving... https://github.com/iiab/calibre-web/pull/125
                         conn.execute("UPDATE media SET path = ? WHERE webpath = ?", (new_video_path, self.media_url))
                         conn.execute("UPDATE media SET webpath = ? WHERE path = ?", (f"{self.media_url}&timestamp={int(datetime.now().timestamp())}", new_video_path))
+                        conn.commit()
```

Comment:
I verified this too: commit() is required if the sqlite3 connection is in autocommit=False mode, which is the recommended value.

But it seems that autocommit =False is not the current default:
 https://docs.python.org/3/library/sqlite3.html#sqlite3.Connection.autocommit

No change required here, but it's probably best to set autocommit=False on all sqlite connections? Otherwise, a crash happening between these two UPDATE statements would leave the database in an inconsistent state.

I also noticed that the `conn.close()` at line 127 can be removed:

```python
with sqlite3.connect(XKLB_DB_FILE) as conn:
    ...transactions done here...
    # conn is implicitly closed when leaving the with block

conn.close()  # this additional close has no effect and can be removed
```

Reply:
> No change required here, but it's probably best to set autocommit=False on all sqlite connections? 

FYI, `sqlite3.connect(... autocommit =False)` requires Python 3.12, so it's probably best to avoid it while there are still users on older Python versions.

Reply:
After reading the docs on how sqlite3.Connection objects interact with context managers, I realize that it doesn't work the same of `file.open()` and other simple I/O objects.

```python
with sqlite3.connect(XKLB_DB_FILE) as conn:
    # ^-- the above context manager will implicitly start an SQL transaction
    ...queries executed here are part of the transaction...
    # The transaction is committed, equivalent to `conn.commit()`

conn.close()  # Still needed because the with block leaves the connection open
```

See: https://docs.python.org/3/library/sqlite3.html#how-to-use-the-connection-context-manager

Reply:
In other words: you don't need to add `conn.commit()` when you're already inside a with block and you will exit the block cleanly (e.g. without throwing an exception or crashing).